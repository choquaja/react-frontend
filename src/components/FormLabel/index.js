import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

const FormLabel = styled.label`
  display: block;
  ${props => props.inline && css`
    display: inline-block;
    margin-right: 1.5rem;
  `}
  font-weight: 700;
  margin-bottom: .5rem;
`;

FormLabel.propTypes = {
  inline: PropTypes.bool,
};

FormLabel.defaultProps = {
  inline: false,
};

export default FormLabel;
