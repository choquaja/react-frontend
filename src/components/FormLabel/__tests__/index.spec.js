import React from 'react';
import { shallow } from 'enzyme';
import FormLabel from '../';

describe('FormLabel', () => {
  it('renders without crashing', () => {
    shallow(<FormLabel />);
  });

  it('renders with inline set true', () => {
    const inline = shallow(<FormLabel inline />);
    expect(inline).toMatchSnapshot();
  });
});
