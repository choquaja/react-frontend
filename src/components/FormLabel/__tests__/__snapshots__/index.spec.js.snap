// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLabel renders with inline set true 1`] = `
<label
  className="FormLabel__FormLabel-eRZfOl kwHyEn"
/>
`;
